### PR TITLE
Integrate SECURITY-2458 changes

### PR DESCRIPTION
**Pull request by daniel-beck**
Context:
https://www.jenkins.io/doc/upgrade-guide/2.303/#SECURITY-2458
https://www.jenkins.io/doc/developer/security/remoting-callables/

Probably debatable whether we want to keep the new behavior to allow `RoleChecker#check(RoleSensitive)`; probably makes sense to amend this and just allow all remoting callables to ignore the role check (not just some). For now, this needs merging so that the released change is on `master` though.

### CODE REVIEW

**Comment by jglick** [comment]
Can you resolve the merge conflict?

File: pom.xml [hunk 1]
```diff
@@ -62,7 +62,7 @@ THE SOFTWARE.
   </scm>
 
   <properties>
-    <revision>4.12</revision>
+    <revision>4.11.2</revision>
```
Comment (jglick) [inline pom.xml:65]: revert please

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
Still needs merge conflict resolved.

**Review by jglick — APPROVED** [review]

**Review by jeffret-b — APPROVED** [review]

File: src/main/java/hudson/remoting/RequiredRoleCheckerWrapper.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.jenkinsci.remoting.Role;
+import org.jenkinsci.remoting.RoleChecker;
+import org.jenkinsci.remoting.RoleSensitive;
+
+import java.util.Collection;
+
+/**
+ * Implementation that can tell whether a {@code #check} method was invoked.
+ */
+/* package-private */ class RequiredRoleCheckerWrapper extends RoleChecker {
```
Comment (jvz) [inline src/main/java/hudson/remoting/RequiredRoleCheckerWrapper.java:35]: This sort of security state checking/reporting seems like a pattern that should be replicated in future patches. I love this idea!

**Review by jvz — APPROVED** [review]

**Comment by daniel-beck** [comment]
@jeffret-b @oleg-nenashev  

Looks like [I messed up the staging](https://issues.jenkins.io/browse/JENKINS-67227) because I forgot this is signed.

Proposal to clean this up towards 2.319.1 (post-RC) and the next weekly release, both of which use remoting 4.11.1:

We could 
1. restore the `v4.11.x` branch,
2. point a [remoting profile in the `jenkins-infra/release` repo](https://github.com/jenkins-infra/release/blob/master/profile.d/components/remoting) to that branch,
3. set the MBPL on release.ci up to use that branch (it currently only checks master),
4. and build it.

I expect we'd get 4.11.2 fairly easily and could still integrate that, assuming the pipeline works as expected.

Thoughts?

CC @cathychan @timja

**Comment by jeffret-b** [comment]
Can't we just release 4.12 and have people use that if they care? Is there any reason we need a 4.11.2 specifically?

We could include #490 or not in that next release.

**Comment by daniel-beck** [comment]
2.319.1 was RC last week, release next week. Ideally we'd not integrate random changes that happen to be on master in LTS.

**Comment by jeffret-b** [comment]
If you can get the build environment to work that way, it's fine.

**Comment by daniel-beck** [comment]
Thanks! Will look into that tomorrow.

**Comment by daniel-beck** [comment]
Had to mess with the branch a bit because conflict resolution happened there. Too bad it was squashed, but we cannot do anything about that now...

**Pull request merged**